sql-private-pool/cloudbuild.yaml

```diff
@@ -0,0 +1,50 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# [START cloudbuild_sql_private_pool]
+steps:
+  - id: "docker-build"
+    name: "gcr.io/cloud-builders/docker"
+    args: ["build", "-t", "${_IMAGE_NAME}", "sql-private-pool/."]
+
+  - id: "docker-push"
```

Comment:
I keep going back and forth on this push.  Part of me thinks this is an unnecessary step and we could build and run the container locally without pushing it.   But, it would look messier and would be harder to follow.   You'd have to convert the nicely written steps to messy bash to build and run in the same step, probably.  There's probably a way of sharing the image between steps, but I haven't ever tried to do it and I think that would get messy as well. 

So, I'm coming down on the side of readability.  Build it, push it, pull it back down, run it.

Reply:
This is the sample where we are following standard operating practices: build, push, do. I had to double-check that this wasn't private-pool, not sql-proxy. 

The only time we wouldn't want to push a container to a registry is when it's specifically ephemeral (which is it is for our temporary image). But I'm also not 100% sure about when we should have a push step vs declaring them in `images` (https://cloud.google.com/build/docs/build-config-file-schema#images)

Reply:
Sounds good!

The `images` config results in a push at the end, so we should do a docker push if we want to use the image in one of the steps.